incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -216,9 +216,6 @@ func (r *HierarchyConfigReconciler) syncWithForest(log logr.Logger, nsInst *core
 	hadCrit := ns.HasLocalCritCondition()
 	ns.ClearConditions()
 
-	// Record whether the namespace is being deleted; this is useful for object validators.
-	ns.IsDeleting = !nsInst.DeletionTimestamp.IsZero()
-
```

Comment:
Can you please delete the field as well? https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/internal/forest/namespace.go#L55

And you also have to update the code in `syncSubnamespaceParent` if you're removing the code that sets the field.

Reply:
i was about to do it before you suggested not to remove [this](https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/internal/reconcilers/hierarchy_config.go#L295-L307)  inside `syncSubnamespaceParent`
If i removed https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/internal/forest/namespace.go#L55, then `syncSubnamespaceParent` will have error since it uses `IsDeleting`.

or am i miss something?

Reply:
Sorry I wasn't clear. In [this comment](https://github.com/kubernetes-sigs/multi-tenancy/pull/1355#discussion_r567532923), I meant that you should replace the check for `ns.IsDeleting` with a check of `nsInst.DeletionTimestamp`, since they have exactly the same information. We need all the _other_ lines in that block of code, but we don't need `ns.IsDeleting` since we have a good alternative.

Does that make sense?

Reply:
ah i get it. Thanks for clarifying.